validation/superchain-version_test.go

```diff
@@ -16,9 +16,52 @@ import (
 	"github.com/ethereum/go-ethereum/ethclient"
 )
 
+var isSemverAcceptable = func(desired, actual string) bool {
+	return desired == actual
+}
+
+// TestSuperchainWideContractVersions will check that
+//   - for each superchain
+//   - for each declared superchain-wide contract "Foo" : version entry in the corresponding superchain's semver.yaml
+//   - the superchain has a contract Foo deployed (at an address declared in superchain.yaml) at the same version
+//
+// Actual semvers are
+// read from the L1 chain RPC provider for the superchain in question.
+func TestSuperchainWideContractVersions(t *testing.T) {
+	for name, superchain := range Superchains {
+		rpcEndpoint := superchain.Config.L1.PublicRPC
+		require.NotEmpty(t, rpcEndpoint)
+
+		client, err := ethclient.Dial(rpcEndpoint)
+		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+		contractNames := []string{
```

Comment:
Do we have to manually specify the contract names like this? I wonder if there's a way for us to combine this test + the one below it, since they both achieve roughly the same thing.

Reply:
[1e52cf0](https://github.com/ethereum-optimism/superchain-registry/pull/73/commits/1e52cf0079b085ff71601425a3fafad83410db58)